actions/bump-and-release-cloud-plugin/action.yaml

```diff
@@ -0,0 +1,320 @@
+      The path to the plugin validator config file. This is used to validate the plugin before release.
+    required: false
+outputs:
+  version:
+    description: The semantic version that the package.json was bumped too
+    value: ${{ steps.output-version.outputs.version }}
+  version-tag:
+    description: The tag that was created for the version
+    value: ${{ steps.output-version.outputs.version-tag }}
+
+runs:
+  using: composite
+  steps:
+    - name: Specify GOPRIVATE environment variable
+      shell: bash
+      run: |
+        echo "GOPRIVATE=github.com/${{ github.repository_owner }}/*" >> $GITHUB_ENV
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        ref: ${{ inputs.default-branch }}
+        token: ${{ inputs.github-token }}
+        fetch-depth: 0
+
+    - name: Setup credentials to access private repositories
+      shell: bash
+      run: git config --global url.https://${{ inputs.github-token }}@github.com/.insteadOf https://github.com/
+
+    - name: Setup Node.js environment
+      uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+      with:
+        node-version: lts/iron
+        check-latest: true
+        cache: ${{ inputs.package-manager }}
+
+    - name: Setup Go environment
+      uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+      with:
+        go-version: ${{ inputs.go-version }}
+
+    - name: Install utilities
+      shell: bash
+      run: sudo apt-get install jq
+
+    - name: Is modern yarn
+      id: is-modern-yarn
+      shell: bash
+      if: inputs.package-manager == 'yarn'
+      run: |
+        version=$(yarn --version)
+
+        if [[ $version =~ ^1 ]]; then
+          echo "modern=false" >> $GITHUB_OUTPUT
+        else
+          echo "modern=true" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Install dependencies
+      if: inputs.package-manager == 'yarn' && steps.is-modern-yarn.outputs.modern == 'true'
+      shell: bash
+      run: yarn install --immutable;
+
+    - name: Install dependencies
+      if: inputs.package-manager == 'yarn' && steps.is-modern-yarn.outputs.modern == 'false'
+      shell: bash
+      run: yarn install --frozen-lockfile;
+
+    - name: Install dependencies
+      if: inputs.package-manager == 'npm'
+      shell: bash
+      run: npm ci;
+
+    - name: Prep package.json
+      shell: bash
+      run: |
+        echo "$(jq 'del(.stableVersion)' package.json)" > package.json
+
+    - name: Bump version (old yarn)
+      if: inputs.package-manager == 'yarn' && steps.is-modern-yarn.outputs.modern == 'false'
+      shell: bash
+      run: |
+        yarn version --no-git-tag-version --${{ inputs.release-level }}
+
+    - name: Bump version
+      id: bump-version
+      if: inputs.package-manager != 'yarn' || steps.is-modern-yarn.outputs.modern == 'true'
+      shell: bash
+      run: |
+        ${{ inputs.package-manager }} version ${{ inputs.release-level }}
```

Comment:
this can be optional? for example in our plugin (grafana-aws-app and cloud-home-app) we bump the version in package.json and update the Changelog in a previous step where we automatically create a PR with both changes to be approved by other team member and then, when this PR is merged, we push the tag to start the release process. 

Our Changelog is updated automatically based on the PR titles merged using conventional commits, with specific format and includes some specific changes not all the things unreleased. For that reason I prefer to have bump and update Changelog optional both.

Reply:
The intention of this action is to do that bump,build,test,release.

What would your goal of using the action be at a higher level? Just the ability to build and test the plugin? or a simpler way to cut a GitHub release and push to GCS?

Reply:
both but not including the version and Changelog update in the same action as mandatory